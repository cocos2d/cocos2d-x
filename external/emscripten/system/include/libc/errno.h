#ifndef __ERRNO_H__
#define __ERRNO_H__

#ifndef __error_t_defined
typedef int error_t;
#define __error_t_defined 1
#endif

#include <sys/errno.h>

#endif /* !__ERRNO_H__ */
