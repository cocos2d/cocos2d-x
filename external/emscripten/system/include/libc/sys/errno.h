/* errno is not a global variable, because that would make using it
   non-reentrant.  Instead, its address is returned by the function
   __errno.  */

#ifndef _SYS_ERRNO_H_
#ifdef __cplusplus
extern "C" {
#endif
#define _SYS_ERRNO_H_

#include <sys/reent.h>

#ifndef _REENT_ONLY
#define errno (*__errno())
extern int *__errno _PARAMS ((void));
#endif

/* Please don't use these variables directly.
   Use strerror instead. */
extern __IMPORT _CONST char * _CONST _sys_errlist[];
extern __IMPORT int _sys_nerr;
#ifdef __CYGWIN__
extern __IMPORT const char * const sys_errlist[];
extern __IMPORT int sys_nerr;
#endif

#define __errno_r(ptr) ((ptr)->_errno)

#define	EPERM 1		/* Not super-user */
#define	ENOENT 2	/* No such file or directory */
#define	ESRCH 3		/* No such process */
#define	EINTR 4		/* Interrupted system call */
#define	EIO 5		/* I/O error */
#define	ENXIO 6		/* No such device or address */
#define	E2BIG 7		/* Arg list too long */
#define	ENOEXEC 8	/* Exec format error */
#define	EBADF 9		/* Bad file number */
#define	ECHILD 10	/* No children */
#define	EAGAIN 11	/* No more processes */
#define	ENOMEM 12	/* Not enough core */
#define	EACCES 13	/* Permission denied */
#define	EFAULT 14	/* Bad address */
#ifdef __LINUX_ERRNO_EXTENSIONS__
#define	ENOTBLK 15	/* Block device required */
#endif
#define	EBUSY 16	/* Mount device busy */
#define	EEXIST 17	/* File exists */
#define	EXDEV 18	/* Cross-device link */
#define	ENODEV 19	/* No such device */
#define	ENOTDIR 20	/* Not a directory */
#define	EISDIR 21	/* Is a directory */
#define	EINVAL 22	/* Invalid argument */
#define	ENFILE 23	/* Too many open files in system */
#define	EMFILE 24	/* Too many open files */
#define	ENOTTY 25	/* Not a typewriter */
#define	ETXTBSY 26	/* Text file busy */
#define	EFBIG 27	/* File too large */
#define	ENOSPC 28	/* No space left on device */
#define	ESPIPE 29	/* Illegal seek */
#define	EROFS 30	/* Read only file system */
#define	EMLINK 31	/* Too many links */
#define	EPIPE 32	/* Broken pipe */
#define	EDOM 33		/* Math arg out of domain of func */
#define	ERANGE 34	/* Math result not representable */
#define	ENOMSG 35	/* No message of desired type */
#define	EIDRM 36	/* Identifier removed */
#ifdef __LINUX_ERRNO_EXTENSIONS__
#define	ECHRNG 37	/* Channel number out of range */
#define	EL2NSYNC 38	/* Level 2 not synchronized */
#define	EL3HLT 39	/* Level 3 halted */
#define	EL3RST 40	/* Level 3 reset */
#define	ELNRNG 41	/* Link number out of range */
#define	EUNATCH 42	/* Protocol driver not attached */
#define	ENOCSI 43	/* No CSI structure available */
#define	EL2HLT 44	/* Level 2 halted */
#endif
#define	EDEADLK 45	/* Deadlock condition */
#define	ENOLCK 46	/* No record locks available */
#ifdef __LINUX_ERRNO_EXTENSIONS__
#define EBADE 50	/* Invalid exchange */
#define EBADR 51	/* Invalid request descriptor */
#define EXFULL 52	/* Exchange full */
#define ENOANO 53	/* No anode */
#define EBADRQC 54	/* Invalid request code */
#define EBADSLT 55	/* Invalid slot */
#define EDEADLOCK 56	/* File locking deadlock error */
#define EBFONT 57	/* Bad font file fmt */
#endif
#define ENOSTR 60	/* Device not a stream */
#define ENODATA 61	/* No data (for no delay io) */
#define ETIME 62	/* Timer expired */
#define ENOSR 63	/* Out of streams resources */
#ifdef __LINUX_ERRNO_EXTENSIONS__
#define ENONET 64	/* Machine is not on the network */
#define ENOPKG 65	/* Package not installed */
#define EREMOTE 66	/* The object is remote */
#endif
#define ENOLINK 67	/* The link has been severed */
#ifdef __LINUX_ERRNO_EXTENSIONS__
#define EADV 68		/* Advertise error */
#define ESRMNT 69	/* Srmount error */
#define	ECOMM 70	/* Communication error on send */
#endif
#define EPROTO 71	/* Protocol error */
#define	EMULTIHOP 74	/* Multihop attempted */
#ifdef __LINUX_ERRNO_EXTENSIONS__
#define	ELBIN 75	/* Inode is remote (not really error) */
#define	EDOTDOT 76	/* Cross mount point (not really error) */
#endif
#define EBADMSG 77	/* Trying to read unreadable message */
#define EFTYPE 79	/* Inappropriate file type or format */
#ifdef __LINUX_ERRNO_EXTENSIONS__
#define ENOTUNIQ 80	/* Given log. name not unique */
#define EBADFD 81	/* f.d. invalid for this operation */
#define EREMCHG 82	/* Remote address changed */
#define ELIBACC 83	/* Can't access a needed shared lib */
#define ELIBBAD 84	/* Accessing a corrupted shared lib */
#define ELIBSCN 85	/* .lib section in a.out corrupted */
#define ELIBMAX 86	/* Attempting to link in too many libs */
#define ELIBEXEC 87	/* Attempting to exec a shared library */
#endif
#define ENOSYS 88	/* Function not implemented */
#ifdef __CYGWIN__
#define ENMFILE 89      /* No more files */
#endif
#define ENOTEMPTY 90	/* Directory not empty */
#define ENAMETOOLONG 91	/* File or path name too long */
#define ELOOP 92	/* Too many symbolic links */
#define EOPNOTSUPP 95	/* Operation not supported on transport endpoint */
#define EPFNOSUPPORT 96 /* Protocol family not supported */
#define ECONNRESET 104  /* Connection reset by peer */
#define ENOBUFS 105	/* No buffer space available */
#define EAFNOSUPPORT 106 /* Address family not supported by protocol family */
#define EPROTOTYPE 107	/* Protocol wrong type for socket */
#define ENOTSOCK 108	/* Socket operation on non-socket */
#define ENOPROTOOPT 109	/* Protocol not available */
#ifdef __LINUX_ERRNO_EXTENSIONS__
#define ESHUTDOWN 110	/* Can't send after socket shutdown */
#endif
#define ECONNREFUSED 111	/* Connection refused */
#define EADDRINUSE 112		/* Address already in use */
#define ECONNABORTED 113	/* Connection aborted */
#define ENETUNREACH 114		/* Network is unreachable */
#define ENETDOWN 115		/* Network interface is not configured */
#define ETIMEDOUT 116		/* Connection timed out */
#define EHOSTDOWN 117		/* Host is down */
#define EHOSTUNREACH 118	/* Host is unreachable */
#define EINPROGRESS 119		/* Connection already in progress */
#define EALREADY 120		/* Socket already connected */
#define EDESTADDRREQ 121	/* Destination address required */
#define EMSGSIZE 122		/* Message too long */
#define EPROTONOSUPPORT 123	/* Unknown protocol */
#ifdef __LINUX_ERRNO_EXTENSIONS__
#define ESOCKTNOSUPPORT 124	/* Socket type not supported */
#endif
#define EADDRNOTAVAIL 125	/* Address not available */
#define ENETRESET 126
#define EISCONN 127		/* Socket is already connected */
#define ENOTCONN 128		/* Socket is not connected */
#define ETOOMANYREFS 129
#ifdef __LINUX_ERRNO_EXTENSIONS__
#define EPROCLIM 130
#define EUSERS 131
#endif
#define EDQUOT 132
#define ESTALE 133
#define ENOTSUP 134		/* Not supported */
#ifdef __LINUX_ERRNO_EXTENSIONS__
#define ENOMEDIUM 135   /* No medium (in tape drive) */
#endif
#ifdef __CYGWIN__
#define ENOSHARE 136    /* No such host or network path */
#define ECASECLASH 137  /* Filename exists with different case */
#endif
#define EILSEQ 138
#define EOVERFLOW 139	/* Value too large for defined data type */
#define ECANCELED 140	/* Operation canceled */
#define ENOTRECOVERABLE 141	/* State not recoverable */
#define EOWNERDEAD 142	/* Previous owner died */
#ifdef __LINUX_ERRNO_EXTENSIONS__
#define ESTRPIPE 143	/* Streams pipe error */
#endif
#define EWOULDBLOCK EAGAIN	/* Operation would block */

#define __ELASTERROR 2000	/* Users can add values starting here */

#ifdef __cplusplus
}
#endif
#endif /* _SYS_ERRNO_H */
