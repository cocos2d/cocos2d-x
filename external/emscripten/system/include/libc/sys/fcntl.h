#ifndef _SYS_FCNTL_H_
#define _SYS_FCNTL_H_
#include <sys/_default_fcntl.h>
#endif
