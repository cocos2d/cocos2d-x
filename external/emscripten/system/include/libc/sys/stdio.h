#ifndef _NEWLIB_STDIO_H
#define _NEWLIB_STDIO_H

#include <sys/lock.h>
#include <sys/reent.h>

/* Internal locking macros, used to protect stdio functions.  In the
   general case, expand to nothing. Use __SSTR flag in FILE _flags to
   detect if FILE is private to sprintf/sscanf class of functions; if
   set then do nothing as lock is not initialised. */
#if !defined(_flockfile)
#ifndef __SINGLE_THREAD__
#  define _flockfile(fp) (((fp)->_flags & __SSTR) ? 0 : __lock_acquire_recursive((fp)->_lock))
#else
#  define _flockfile(fp)	(_CAST_VOID 0)
#endif
#endif

#if !defined(_funlockfile)
#ifndef __SINGLE_THREAD__
#  define _funlockfile(fp) (((fp)->_flags & __SSTR) ? 0 : __lock_release_recursive((fp)->_lock))
#else
#  define _funlockfile(fp)	(_CAST_VOID 0)
#endif
#endif

#endif /* _NEWLIB_STDIO_H */
