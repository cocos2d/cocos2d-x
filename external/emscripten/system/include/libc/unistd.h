#ifndef _UNISTD_H_
#define _UNISTD_H_

# include <sys/unistd.h>

#endif /* _UNISTD_H_ */
