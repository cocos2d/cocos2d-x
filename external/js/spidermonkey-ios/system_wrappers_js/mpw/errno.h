#pragma GCC system_header
#pragma GCC visibility push(default)
#include_next <mpw/errno.h>
#pragma GCC visibility pop
