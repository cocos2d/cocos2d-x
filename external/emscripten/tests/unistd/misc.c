#include <stdio.h>
#include <errno.h>
#include <unistd.h>
#include <fcntl.h>

int main() {
  int f = open("/", O_RDONLY);

  sync();

  printf("fsync(good): %d", fsync(f));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("fsync(bad): %d", fsync(42));
  printf(", errno: %d\n", errno);
  errno = 0;

  printf("fdatasync(good): %d", fdatasync(f));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("fdatasync(bad): %d", fdatasync(42));
  printf(", errno: %d\n", errno);
  errno = 0;

  printf("tcgetpgrp(good): %d", tcgetpgrp(f));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("tcgetpgrp(bad): %d", tcgetpgrp(42));
  printf(", errno: %d\n", errno);
  errno = 0;

  printf("tcsetpgrp(good): %d", tcsetpgrp(f, 123));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("tcsetpgrp(bad): %d", tcsetpgrp(42, 123));
  printf(", errno: %d\n", errno);
  errno = 0;

  printf("link: %d", link("/here", "/there"));
  printf(", errno: %d\n", errno);
  errno = 0;

  printf("lockf(good): %d", lockf(f, 123, 456));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("lockf(bad): %d", lockf(42, 123, 456));
  printf(", errno: %d\n", errno);
  errno = 0;

  printf("nice: %d", nice(42));
  printf(", errno: %d\n", errno);
  errno = 0;

  printf("pause: %d", pause());
  printf(", errno: %d\n", errno);
  errno = 0;

  int pipe_arg[2];
  printf("pipe(good): %d", pipe(pipe_arg));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("pipe(bad): %d", pipe(0));
  printf(", errno: %d\n", errno);
  errno = 0;

  char* exec_argv[] = {"arg", 0};
  char* exec_env[] = {"a=b", 0};
  printf("execl: %d", execl("/program", "arg", 0));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("execle: %d", execle("/program", "arg", 0, exec_env));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("execlp: %d", execlp("program", "arg", 0));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("execv: %d", execv("program", exec_argv));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("execve: %d", execve("program", exec_argv, exec_env));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("execvp: %d", execvp("program", exec_argv));
  printf(", errno: %d\n", errno);
  errno = 0;

  printf("chown(good): %d", chown("/", 123, 456));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("chown(bad): %d", chown("/noexist", 123, 456));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("lchown(good): %d", lchown("/", 123, 456));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("lchown(bad): %d", lchown("/noexist", 123, 456));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("fchown(good): %d", fchown(f, 123, 456));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("fchown(bad): %d", fchown(42, 123, 456));
  printf(", errno: %d\n", errno);
  errno = 0;

  printf("alarm: %d", alarm(42));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("ualarm: %ld", ualarm(123, 456));
  printf(", errno: %d\n", errno);
  errno = 0;

  printf("fork: %d", fork());
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("vfork: %d", vfork());
  printf(", errno: %d\n", errno);
  errno = 0;

  printf("crypt: %s", crypt("foo", "bar"));
  printf(", errno: %d\n", errno);
  errno = 0;
  char encrypt_block[64] = {0};
  printf("encrypt");
  encrypt(encrypt_block, 42);
  printf(", errno: %d\n", errno);
  errno = 0;

  printf("getgid: %d", getgid());
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("getegid: %d", getegid());
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("getuid: %d", getuid());
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("geteuid: %d", geteuid());
  printf(", errno: %d\n", errno);
  errno = 0;

  printf("getpgrp: %d", getpgrp());
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("getpid: %d", getpid());
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("getppid: %d", getppid());
  printf(", errno: %d\n", errno);
  errno = 0;

  printf("getpgid: %d", getpgid(42));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("getsid: %d", getsid(42));
  printf(", errno: %d\n", errno);
  errno = 0;

  printf("setgid: %d", setgid(42));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("setegid: %d", setegid(42));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("setuid: %d", setuid(42));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("seteuid: %d", seteuid(42));
  printf(", errno: %d\n", errno);
  errno = 0;

  printf("setpgrp: %d", setpgrp());
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("setsid: %d", setsid());
  printf(", errno: %d\n", errno);
  errno = 0;

  printf("setpgid: %d", setpgid(123, 456));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("setregid: %d", setregid(123, 456));
  printf(", errno: %d\n", errno);
  errno = 0;
  printf("setreuid: %d", setreuid(123, 456));
  printf(", errno: %d\n", errno);
  errno = 0;

  gid_t groups[10] = {42};
  printf("getgroups: %d", getgroups(10, groups));
  printf(", result: %d", groups[0]);
  printf(", errno: %d\n", errno);
  errno = 0;

  gid_t groups2[1] = {0};
  printf("setgroups: %d", setgroups(1, groups2));
  printf(", errno: %d\n", errno);
  errno = 0;

  return 0;
}
