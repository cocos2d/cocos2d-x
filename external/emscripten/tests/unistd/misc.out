fsync(good): 0, errno: 0
fsync(bad): -1, errno: 9
fdatasync(good): 0, errno: 0
fdatasync(bad): -1, errno: 9
tcgetpgrp(good): 0, errno: 0
tcgetpgrp(bad): 0, errno: 0
tcsetpgrp(good): -1, errno: 22
tcsetpgrp(bad): -1, errno: 22
link: -1, errno: 31
lockf(good): 0, errno: 0
lockf(bad): -1, errno: 9
nice: 0, errno: 1
pause: -1, errno: 4
pipe(good): -1, errno: 38
pipe(bad): -1, errno: 38
execl: -1, errno: 8
execle: -1, errno: 8
execlp: -1, errno: 8
execv: -1, errno: 8
execve: -1, errno: 8
execvp: -1, errno: 8
chown(good): 0, errno: 0
chown(bad): -1, errno: 2
lchown(good): 0, errno: 0
lchown(bad): -1, errno: 2
fchown(good): 0, errno: 0
fchown(bad): -1, errno: 9
alarm: 0, errno: 0
ualarm: 0, errno: 0
fork: -1, errno: 11
vfork: -1, errno: 11
crypt: (null), errno: 38
encrypt, errno: 38
getgid: 0, errno: 0
getegid: 0, errno: 0
getuid: 0, errno: 0
geteuid: 0, errno: 0
getpgrp: 0, errno: 0
getpid: 0, errno: 0
getppid: 0, errno: 0
getpgid: 0, errno: 0
getsid: 0, errno: 0
setgid: -1, errno: 1
setegid: -1, errno: 1
setuid: -1, errno: 1
seteuid: -1, errno: 1
